gc/base/GlobalCollector.hpp

```diff
@@ -36,9 +37,23 @@ class MM_MemorySubSpace;
 class MM_GlobalCollector : public MM_Collector {
 private:
 protected:
+	MM_GlobalCollectorDelegate _delegate; /**< Language specific delegate -- subclass must initialize */
+
+public:
+
+private:
+protected:
 	virtual void internalPostCollect(MM_EnvironmentBase *env, MM_MemorySubSpace *subSpace);
 
 public:
+	MM_GlobalCollectorDelegate *getGlobalCollectorDelegate() { return &_delegate; }
+
+	/**
+	 * Informational method called when master thread is started, to allow language-specific action is required. Calls
+	 * MM_GlobalCollectorDelegate::startMasterThread() to effect language specialization.
+	 */
+	virtual void startMasterThread(MM_EnvironmentBase *env) { _delegate.startMasterThread(env); }
```

Comment:
This is only used for Global collectors?  Generational and Incremental collectors do not need this functionality? This feels like something that may belong in a different class and not the collector hierarchy.